### PR TITLE
Add Array#select drill

### PR DESCRIPTION
**Pull request by bmurphy1**
Added an Array#select drill for #4. Feedback welcome, this is a great tool for beginners. :)

### CODE REVIEW

File: lib/ruby_drills/array/select_drill.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class SelectDrill < Drill
+
+  def setup
+    @values = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]
+    @hints = ["Are you being selective enough?",
+              "This method is called filter in other languages.",
+              "http://ruby-doc.org//core-2.2.0/Enumerable.html#method-i-select"]
```
Comment (dskecse) [inline lib/ruby_drills/array/select_drill.rb]: This should be "http://ruby-doc.org/core/Enumerable.html#method-i-select" instead.

File: lib/ruby_drills/array/select_drill.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def show
+    puts %{
```
Comment (dskecse) [inline lib/ruby_drills/array/select_drill.rb]: You'd better use a squiggly heredoc `<<~` introduced in Ruby 2.3 https://infinum.co/the-capsized-eight/articles/multiline-strings-ruby-2-3-0-the-squiggly-heredoc.

File: lib/ruby_drills/array/select_drill.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+  end
+
+  def reference
+    "values.select {|x| x % 3 == 0}"
+  end
+
+  def valid?(input)
+    ['select', 'find_all'].any? { |x| input.include?(x) }
```
Comment (dskecse) [inline lib/ruby_drills/array/select_drill.rb]: In fact, you don't need to use `any?` here. Rewrite it like this `%w(select find_all).include?(x)`.

File: lib/ruby_drills/array/select_drill.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+divisible by 3.
+
+Use the method that returns an array containing all
+elements for which the given block returns a true value.
+    }
+
+  end
+
+  def reference
+    "values.select {|x| x % 3 == 0}"
```
Comment (saslani) [inline lib/ruby_drills/array/select_drill.rb]: should be "@values.select {|x| x % 3 == 0}"

**Comment by saslani** [comment]
I realize that this PR was from 2 years ago. I Truly appreciate your time on this PR. I will merge this in and fix the small issue to get it working

**Pull request merged**